google/cloud/testing_util/assert_ok.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_ASSERT_OK_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_ASSERT_OK_H_
+
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <gtest/gtest.h>
+
+namespace testing {
+namespace internal {
+
+// A unary predicate-formatter for google::cloud::Status.
+testing::AssertionResult IsOkPredFormat(char const* expr,
+                                        ::google::cloud::Status const& status);
+
+// A unary predicate-formatter for google::cloud::StatusOr<T>.
+template <typename T>
+testing::AssertionResult IsOkPredFormat(
+    char const* expr, ::google::cloud::StatusOr<T> const& status_or) {
+  if (status_or) {
+    return testing::AssertionSuccess();
+  }
+  return IsOkPredFormat(expr, status_or.status());
+}
+
+}  // namespace internal
+}  // namespace testing
+
+#define ASSERT_STATUS_OK(val) \
```

Comment:
It seems like we need a story for ASSERT/EXPECT_STATUS_NOT_OK.  Abseil leaves this to EXPECT_THAT.

  EXPECT_THAT(status, IsOk())
  EXPECT_THAT(status, IsNotOk())

Reply:
Currently the code uses:
EXPECT_FALSE(status.ok())
EXPECT_FALSE(status_or)

The positive counterpart brought value because it was printing the status. I can add those if you insist, but it seems redundant to me, TBH (longer to type, no additional debugging information).

Reply:
Right you are.  Thanks.  But I was thinking of something more like expecting a particular (set of) non-OK value(s), which is what those EXPECT_FALSE()s should probably be more like, but I'm happy to defer consideration of such a feature.

Reply:
Got it, I didn't think about it. I filed #1999 so that we don't forget about it.